lib/doorkeeper/openid_connect/response_mode.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module OpenidConnect
+    class ResponseMode
+      attr_reader :type
+
+      def initialize(response_type)
+        @type = response_type
+      end
+
+      def fragment?
+        mode == 'fragment'
+      end
+
+      def query?
+        mode == 'query'
+      end
+
+      def mode
+        case type
+        when 'token', 'id_token', 'id_token token'
```

Comment:
Trying to double-check the specs here:

- `id_token` and `id_token token` are used for OIDC Implicit Flow which mandates `fragment`: https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthRequest
- `token` is used in normal OAuth and https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#ResponseModes specifies `fragment` as the default.

So I think this looks correct! :+1:

Reply:
Yea I think it should be according to spec, nice that you double checked. Note that I think that I should try and upstream the response_mode as part of the new ”Grant Flows” in doorkeeper (since it makes more sense that they dictate what is default and/or valid response modes. 

But I think this is an ok change until it is available from there.